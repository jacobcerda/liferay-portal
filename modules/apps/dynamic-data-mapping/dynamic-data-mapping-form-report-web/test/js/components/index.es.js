/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, render} from '@testing-library/react';
import React from 'react';

import Index from '../../../src/main/resources/META-INF/resources/js/index.es';

const props = {
	data: JSON.stringify('{field1: {values: {option1: 1}, type:"radio"}'),
	fields: [
		{name: 'field1', type: 'radio'},
		{name: 'field2', type: 'radio'},
		{name: 'field3', type: 'radio'},
	],
};

describe('index', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {asFragment} = render(<Index {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('renders nothing when there is no data', () => {
		const {asFragment} = render(<Index {...props} data={null} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
