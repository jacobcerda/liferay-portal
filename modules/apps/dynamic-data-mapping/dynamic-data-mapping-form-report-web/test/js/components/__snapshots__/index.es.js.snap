// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders 1`] = `
<DocumentFragment>
  <div
    class="form-report"
  >
    <div
      class="col-md-8"
    >
      <div
        class="sheet"
      >
        <div
          class="col-md-12"
        >
          <div
            class="card card-type-asset form-check form-check-card form-check-top-left image-card"
          >
            <div
              class="card-header card-item-first aspect-ratio"
            >
              <div
                class="aspect-ratio-item aspect-ratio-item-center-left aspect-ratio-item-fluid card-symbol card-type-asset-icon"
                data-tooltip-align="bottom"
                data-tooltip-delay="300"
                title="radio-field-type-label"
              >
                <svg
                  class="lexicon-icon lexicon-icon-radio-button"
                  role="presentation"
                >
                  <use
                    xlink:href="#radio-button"
                  />
                </svg>
              </div>
              <div
                class="field-info"
              >
                <p
                  class="card-title"
                  title="field1"
                >
                  <span
                    class="text-truncate-inline"
                  >
                    <span
                      class="text-truncate"
                    >
                      field1
                    </span>
                  </span>
                </p>
                <p
                  class="card-text"
                  title="there-are-no-entries"
                >
                  there-are-no-entries
                </p>
              </div>
            </div>
            <div
              class="card-body"
            >
              <div
                class="empty-message taglib-empty-result-message"
              >
                <div
                  class="taglib-empty-result-message-header"
                />
                <div
                  class="sheet-text text-center text-muted"
                >
                  <h1
                    class="text-default"
                  >
                    there-are-no-entries
                  </h1>
                  <p
                    class="text-default"
                  >
                    entries-submitted-with-this-field-filled-will-show-up-here
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-md-8"
    >
      <div
        class="sheet"
      >
        <div
          class="col-md-12"
        >
          <div
            class="card card-type-asset form-check form-check-card form-check-top-left image-card"
          >
            <div
              class="card-header card-item-first aspect-ratio"
            >
              <div
                class="aspect-ratio-item aspect-ratio-item-center-left aspect-ratio-item-fluid card-symbol card-type-asset-icon"
                data-tooltip-align="bottom"
                data-tooltip-delay="300"
                title="radio-field-type-label"
              >
                <svg
                  class="lexicon-icon lexicon-icon-radio-button"
                  role="presentation"
                >
                  <use
                    xlink:href="#radio-button"
                  />
                </svg>
              </div>
              <div
                class="field-info"
              >
                <p
                  class="card-title"
                  title="field2"
                >
                  <span
                    class="text-truncate-inline"
                  >
                    <span
                      class="text-truncate"
                    >
                      field2
                    </span>
                  </span>
                </p>
                <p
                  class="card-text"
                  title="there-are-no-entries"
                >
                  there-are-no-entries
                </p>
              </div>
            </div>
            <div
              class="card-body"
            >
              <div
                class="empty-message taglib-empty-result-message"
              >
                <div
                  class="taglib-empty-result-message-header"
                />
                <div
                  class="sheet-text text-center text-muted"
                >
                  <h1
                    class="text-default"
                  >
                    there-are-no-entries
                  </h1>
                  <p
                    class="text-default"
                  >
                    entries-submitted-with-this-field-filled-will-show-up-here
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-md-8"
    >
      <div
        class="sheet"
      >
        <div
          class="col-md-12"
        >
          <div
            class="card card-type-asset form-check form-check-card form-check-top-left image-card"
          >
            <div
              class="card-header card-item-first aspect-ratio"
            >
              <div
                class="aspect-ratio-item aspect-ratio-item-center-left aspect-ratio-item-fluid card-symbol card-type-asset-icon"
                data-tooltip-align="bottom"
                data-tooltip-delay="300"
                title="radio-field-type-label"
              >
                <svg
                  class="lexicon-icon lexicon-icon-radio-button"
                  role="presentation"
                >
                  <use
                    xlink:href="#radio-button"
                  />
                </svg>
              </div>
              <div
                class="field-info"
              >
                <p
                  class="card-title"
                  title="field3"
                >
                  <span
                    class="text-truncate-inline"
                  >
                    <span
                      class="text-truncate"
                    >
                      field3
                    </span>
                  </span>
                </p>
                <p
                  class="card-text"
                  title="there-are-no-entries"
                >
                  there-are-no-entries
                </p>
              </div>
            </div>
            <div
              class="card-body"
            >
              <div
                class="empty-message taglib-empty-result-message"
              >
                <div
                  class="taglib-empty-result-message-header"
                />
                <div
                  class="sheet-text text-center text-muted"
                >
                  <h1
                    class="text-default"
                  >
                    there-are-no-entries
                  </h1>
                  <p
                    class="text-default"
                  >
                    entries-submitted-with-this-field-filled-will-show-up-here
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`index renders nothing when there is no data 1`] = `<DocumentFragment />`;
